test/test_suite.sh

```diff
@@ -1,253 +1,276 @@
 #!./test/libs/bats/bin/bats
 
 load 'libs/bats-support/load'
-# load 'libs/bats-assert/load'
-
-@test "Version" {
-  run bash -c 'echo ">version" | nc -v 127.0.0.1 4711'
-  echo "output: ${lines[@]}"
-  [[ ${lines[0]} == "Connection to 127.0.0.1 4711 port [tcp/*] succeeded!" ]]
-  [[ ${lines[1]} =~ "version" ]]
-  [[ ${lines[2]} =~ "tag" ]]
-  [[ ${lines[3]} =~ "branch" ]]
-  [[ ${lines[4]} =~ "date" ]]
-  [[ ${lines[5]} == "---EOM---" ]]
-}
-
-@test "Statistics" {
-  run bash -c 'echo ">stats" | nc -v 127.0.0.1 4711'
-  echo "output: ${lines[@]}"
-  [[ ${lines[0]} == "Connection to 127.0.0.1 4711 port [tcp/*] succeeded!" ]]
-  [[ ${lines[1]} =~ "domains_being_blocked -1" ]]
-  [[ ${lines[2]} =~ "dns_queries_today 7" ]]
-  [[ ${lines[3]} =~ "ads_blocked_today 2" ]]
-  [[ ${lines[4]} =~ "ads_percentage_today 28.571428" ]]
-  [[ ${lines[5]} =~ "unique_domains 6" ]]
-  [[ ${lines[6]} =~ "queries_forwarded 3" ]]
-  [[ ${lines[7]} =~ "queries_cached 2" ]]
-  [[ ${lines[8]} == "clients_ever_seen 3" ]]
-  [[ ${lines[9]} == "unique_clients 3" ]]
-  [[ ${lines[10]} == "status unknown" ]]
-  [[ ${lines[11]} == "---EOM---" ]]
+
+@test "Version, Tag, Branch, Hash, Date is reported" {
+  run bash -c 'echo ">version >quit" | nc -v 127.0.0.1 4711'
+  printf "%s\n" "${lines[@]}"
+  [[ ${lines[1]} == "version "* ]]
+  [[ ${lines[2]} == "tag "* ]]
+  [[ ${lines[3]} == "branch "* ]]
+  [[ ${lines[4]} == "hash "* ]]
+  [[ ${lines[5]} == "date "* ]]
+  [[ ${lines[6]} == "" ]]
+}
+
+@test "Blacklisted domain is blocked" {
+  run bash -c "dig blacklisted.com @127.0.0.1 +short"
+  printf "%s\n" "${lines[@]}"
+  [[ ${lines[0]} == "0.0.0.0" ]]
+  [[ ${lines[1]} == "" ]]
```

Comment:
What is the significance of the second line being blank?

Reply:
I added this everywhere to exclude that there is *more* output than just this. Example in this case would be if there are several `A` records for this domain. Try, e.g.,
```
$ dig ebay.de +short
66.135.196.249
66.135.195.175
```
We want the test to fail if **more** than only `0.0.0.0` is returned.

Reply:
Would something like this work better?
```bash
output=$(dig blacklisted.com @127.0.0.1 +short)
{
    echo "0.0.0.0"
}>expected
[[ "${output}" == "${expected}" ]]
```

Reply:
This makes sure that there is no more output beyond the last empty line.

Reply:
This would be possible as well. I wanted to use the same syntax everywhere. The current way of checking each line individually has the advantage that `bats` will tell us exactly which line isn't equal lessening the work when comparing the outputs. This is obviously no issue here, but, e.g., for the Top Lists tests.

Reply:
Will bats output the data when the test case fails? If so, then it shouldn't be that big of a deal.

Reply:
Yes, `bats` will output the data but we will need to check manually which lines mismatch. Again, this is more a question of if we want to use the same checking mechanism in all tests.

Reply:
The outputs are all rather short. We should try to keep the mechanism the same (although exceptions can be made), so this is more a question of if this method is better than the other.

Reply:
The method you suggested wouldn't work for `@test "pihole-FTL.db schema as expected"`. I vote for keeping what we already have.

Reply:
That one is an exception, which is allowed.

We can keep what we have, but we just need to remember that the tests could miss output after the empty line.

Reply:
Ah, `bats` doesn't allow empty lines. If the output is
```
A

B




C
```
Then we will have
```
output[0] == "A"
output[1] == "B"
output[2] == "C"
output[3] == ""
```